### DISCUSSION TITLE
Would you be open to having a collaborator on this repository?

### DISCUSSION THREAD

**Discussion by artburkart** [tycrek/degoogle#502] [General]
Hi 👋 

I have referenced this guide several times over the past years, and I've always thought it a helpful resource. I see it's not received an update in nearly a year. Would you be open to having a collaborator on it to handle the submitted issues that have come through?

Best!
Arthur

**Comment by tycrek** [comment]
Hi there! This repo did get a bit overwhelming to maintain, some assistance would be appreciated.

I would want to set up some maintainer guidelines to ensure the repo remains operating the same way, if that's fine.

**Reply by artburkart** [reply]
Thank you for your response. I understand the importance of having clear maintainer guidelines, especially for a repository like this one. At a high level, I think the [Contributor Covenant](https://www.contributor-covenant.org/version/2/1/code_of_conduct/) is a good starting place. A lot of repositories adopt it (i.e., [sindresorhus/awesome](https://github.com/sindresorhus/awesome/blob/efc8cefb820d47fa818d192c42ebd98c1bbdc407/code-of-conduct.md#contributor-covenant-code-of-conduct)). For this project, however, there's probably some thought process you've gone through for each addition or subtraction that would benefit from broader illumination if you're to allow co-maintainers to assist in the future.
Happy to continue discussing it. 🙌

**Reply by tycrek** [reply]
Been a busy week, here's about what I've got so far, in regards to accepting submissions via PR:

- **Rule 0: Never merge any PR that directly edits the README.md file.**
- remain objective and factual when considering a submission
- claims should be backed up by evidence
- opinions should be respected but not considered as evidence
- submissions by project owners of their own projects should be more carefully considered, especially in the case of for-profit projects
- any PR that wants to modify the build script should be verified by **me** (tycrek) directly.

General advice:

- for mobile apps, when source code is available, ensure that the repo is reasonably up to date with currently supported OS versions and support schedules. For example, it may not be a great idea to accept an app that hasn’t received an update in 8 years.
- The [Useful-Forks](https://useful-forks.github.io/) project can help maintainers find updated forks of projects.
- Users are welcome to submit via Issues, and maintainers should be the ones to open a PR if a user has not. Not every GitHub user is a developer comfortable with using Pull Requests.

Not the fanciest but it's been a while since I've merged or added anything. Might be worth adding this to a wiki page as well but I'd love to hear your thoughts so far.

I'd also love to implement the Contributor Covenant. Since it includes Attribution, would it be as simple as copying it into a new file?

Lastly, pinging @danarel as I forgot you're listed as a maintainer already! Are you still interested in this role? No worries if not :)

**Reply by artburkart** [reply]
I love this.

**Reply by artburkart** [reply]
> I'd also love to implement the Contributor Covenant. Since it includes Attribution, would it be as simple as copying it into a new file?

Ah, I missed this question. Yeah, you'd just drop something like this at the bottom of your `CODE_OF_CONDUCT.md` file:

```md
## Attribution

This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 1.4,
available at https://www.contributor-covenant.org/version/1/4/code-of-conduct.html

[homepage]: https://www.contributor-covenant.org
```

**Reply by artburkart** [reply]
This page shows adopters: https://www.contributor-covenant.org/adopters/
All the links lead to repositories. I spot checked a dozen or so and couldn't find an example that didn't attribute to the covenant in a file called `CODE_OF_CONDUCT.md`.

**Comment by danarel** [comment]
Count me in. ThinkPrivacy is coming to close in a few weeks and I would love to continue helping keep this maintained.

**Reply by freddy-m** [reply]
I'm happy to help out too - in whatever capacity that may be.

**Comment by tycrek** [comment]
Still getting to the Code of Conduct but some updates from today (see #518 for commits)

- Added branch protections to `master` (PR/Workflow/approval required for merge)
- Build script now uses a dedicated bot account @degoogle-bot which has the ability to bypass branch protections on `master`
- Set up a https://github.com/tycrek/degoogle/blob/master/.github/CODEOWNERS file to further protect certain files
- Set up a Workflow to block PR's if they edit the README file (https://github.com/tycrek/degoogle/blob/master/.github/workflows/readme-protection.yml)
- Minor updates to repo meta (https://github.com/tycrek/degoogle/blob/master/CONTRIBUTING.md, build script, Issue/PR templates)

**Reply by artburkart** [reply]
I didn't know about https://anarchy-droid.com until I saw that PR.
That was a hefty amount of effort you put into that PR.

**Reply by tycrek** [reply]
Busy day haha. Last two things, how do these look:

- https://github.com/tycrek/degoogle/wiki/Maintainer-Guidelines
- #529

**Reply by artburkart** [reply]
Hey, I'm really sorry. I thought I'd replied here, but I guess I hadn't. I was going to say those look good.
